go.mod

```diff
@@ -163,35 +178,37 @@ require (
 	github.com/mxk/go-flowrate v0.0.0-20140419014527-cca7078d478f // indirect
 	github.com/peterbourgon/diskv v2.0.1+incompatible // indirect
 	github.com/pkg/errors v0.9.1
-	github.com/pmezard/go-difflib v1.0.0 // indirect
+	github.com/pmezard/go-difflib v1.0.1-0.20181226105442-5d4384ee4fb2 // indirect
 	github.com/prometheus/client_model v0.6.1 // indirect
-	github.com/prometheus/procfs v0.15.0 // indirect
+	github.com/prometheus/procfs v0.15.1 // indirect
 	github.com/russross/blackfriday/v2 v2.1.0 // indirect
 	github.com/sirupsen/logrus v1.9.3 // indirect
 	github.com/tsaarni/x500dn v1.0.0 // indirect
 	github.com/xlab/treeprint v1.2.0 // indirect
 	go.opentelemetry.io/otel/sdk v1.27.0
 	go.opentelemetry.io/otel/trace v1.27.0 // indirect
-	go.starlark.net v0.0.0-20230525235612-a134d8f9ddca // indirect
+	go.starlark.net v0.0.0-20240520160348-046347dcd104 // indirect
 	go.uber.org/multierr v1.11.0 // indirect
 	golang.org/x/mod v0.18.0 // indirect
 	golang.org/x/net v0.26.0
-	golang.org/x/oauth2 v0.20.0 // indirect
+	golang.org/x/oauth2 v0.21.0 // indirect
 	golang.org/x/sync v0.7.0 // indirect
 	golang.org/x/term v0.21.0 // indirect
 	golang.org/x/text v0.16.0 // indirect
 	golang.org/x/time v0.5.0 // indirect
 	golang.org/x/tools v0.22.0 // indirect
 	gomodules.xyz/jsonpatch/v2 v2.4.0 // indirect
-	google.golang.org/genproto/googleapis/api v0.0.0-20240520151616-dc85e6b867a5 // indirect
-	google.golang.org/genproto/googleapis/rpc v0.0.0-20240521202816-d264139d666e // indirect
+	google.golang.org/genproto/googleapis/api v0.0.0-20240528184218-531527333157 // indirect
+	google.golang.org/genproto/googleapis/rpc v0.0.0-20240528184218-531527333157 // indirect
 	gopkg.in/inf.v0 v0.9.1 // indirect
 	gopkg.in/yaml.v2 v2.4.0 // indirect
 	k8s.io/component-base v0.30.1 // indirect
 	k8s.io/klog/v2 v2.120.1 // indirect
-	k8s.io/kube-openapi v0.0.0-20240423202451-8948a665c108 // indirect
+	k8s.io/kube-openapi v0.0.0-20240521193020-835d969ad83a // indirect
 	sigs.k8s.io/json v0.0.0-20221116044647-bc3834ca7abd // indirect
-	sigs.k8s.io/kustomize/api v0.13.5-0.20230601165947-6ce0bf390ce3 // indirect
-	sigs.k8s.io/kustomize/kyaml v0.14.3-0.20230601165947-6ce0bf390ce3 // indirect
+	sigs.k8s.io/kustomize/api v0.17.2 // indirect
+	sigs.k8s.io/kustomize/kyaml v0.17.1 // indirect
 	sigs.k8s.io/structured-merge-diff/v4 v4.4.1 // indirect
 )
+
+replace dario.cat/mergo => github.com/imdario/mergo v0.3.12
```

Comment:
why is this needed ?

Reply:
It's because of incompatible changes between mergo@v1.0.0 and mergo@v0.3.x

go mod verify
go: github.com/imdario/mergo@v1.0.0: parsing go.mod:
        module declares its path as: dario.cat/mergo
                but was required as: github.com/imdario/mergo

go mod graph|grep " github.com/imdario/mergo"
github.com/envoyproxy/gateway github.com/imdario/mergo@v1.0.0
github.com/Masterminds/sprig/v3@v3.2.3 github.com/imdario/mergo@v0.3.11
github.com/rubenv/sql-migrate@v1.6.1 github.com/imdario/mergo@v0.3.13
helm.sh/helm/v3@v3.15.1 github.com/imdario/mergo@v0.3.13
k8s.io/apiextensions-apiserver@v0.30.1 github.com/imdario/mergo@v0.3.6
k8s.io/apiserver@v0.30.1 github.com/imdario/mergo@v0.3.6
k8s.io/cli-runtime@v0.30.1 github.com/imdario/mergo@v0.3.6
k8s.io/client-go@v0.30.1 github.com/imdario/mergo@v0.3.6
k8s.io/kubectl@v0.30.1 github.com/imdario/mergo@v0.3.6
sigs.k8s.io/controller-runtime@v0.18.4 github.com/imdario/mergo@v0.3.6
sigs.k8s.io/gateway-api@v1.1.0 github.com/imdario/mergo@v0.3.16
k8s.io/client-go@v0.18.4 github.com/imdario/mergo@v0.3.5
sigs.k8s.io/controller-runtime@v0.6.1 github.com/imdario/mergo@v0.3.9
k8s.io/client-go@v0.18.2 github.com/imdario/mergo@v0.3.5